crates/runtime/src/guest_type.rs

```diff
@@ -40,6 +40,19 @@ pub trait GuestType<'a>: Sized {
     fn write(ptr: &GuestPtr<'_, Self>, val: Self) -> Result<(), GuestError>;
 }
 
+/// A trait for `GuestType`s that have the same representation in guest memory
+/// as in Rust. These types can be used with the `GuestPtr::as_raw` method to
+/// view as a slice.
+pub trait GuestTypeTransparent<'a>: GuestType<'a> {
```

Comment:
This may want to be an `unsafe` trait as well to codify the guarantees of `as_raw` to guarantee that the `*mut [T]` is valid so long as other pointer/aliasing requirements are upheld

Reply:
Good idea, added `unsafe` and docs explaining why.